applications/crossbar/src/crossbar_maintenance.erl

```diff
@@ -957,6 +971,51 @@ create_app(AppPath, MetaData, MasterAccountDb) ->
                      ,[kzd_app:name(MetaData), MasterAccountDb, _E])
     end.
 
+-spec maybe_update_app(file:filename_all(), kz_json:object(), kz_term:ne_binaries(), kz_term:ne_binary(), kz_json:object()) -> 'no_return'.
+maybe_update_app(AppPath, MetaData, Keys, MasterAccountDb, AppJObj) ->
+    CurrentDocId = kz_doc:id(AppJObj),
+    case update_app_updates(Keys, MetaData, AppJObj) of
+        [] -> io:format(" no metadata changes for app ~s~n", [CurrentDocId]);
+        Updates ->
+            case kz_datamgr:save_doc(MasterAccountDb, kz_json:set_values(Updates, AppJObj)) of
```

Comment:
I would maybe put this in its own function `save_app_updates` or something, avoid the extra nesting.